main.go

```diff
@@ -160,6 +201,46 @@ func realMain(args []string) int {
 	return exit
 }
 
+func convertToJson(path string) ([]byte, error) {
+	values, err := ReadValuesFile(path)
+	if err != nil {
+		//return errors.Wrap(err, "unable to parse YAML")
+		return []byte{}, err
+	}
+	valuesData, err := yaml.Marshal(values)
+	if err != nil {
+		return []byte{}, err
+	}
+	valuesJSON, err := yaml.YAMLToJSON(valuesData)
+	if err != nil {
+		return []byte{}, err
+	}
+	if bytes.Equal(valuesJSON, []byte("null")) {
+		valuesJSON = []byte("{}")
+	}
+	return valuesJSON, err
+}
+
+type Values map[string]interface{}
+
+// ReadValues will parse YAML byte data into a Values.
+func ReadValues(data []byte) (vals Values, err error) {
+	err = yaml.Unmarshal(data, &vals)
+	if len(vals) == 0 {
+		vals = Values{}
+	}
+	return vals, err
+}
+
+// ReadValuesFile will parse a YAML file into a map of values.
+func ReadValuesFile(filename string) (Values, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return map[string]interface{}{}, err
+	}
+	return ReadValues(data)
+}
+
```

Comment:
Looks like most of this was copied from Helm, however, none of this should be needed. All that's required is reading the file into a `[]byte` buffer and then passing that directly into `yaml.YAMLToJSON` to get back the converted `[]byte`s. Something like the following should do the trick.

```go
buf, err := ioutil.ReadFile(path)
if err != nil {
	return nil, err
}
return yaml.YAMLToJSON(buf)
```

Reply:
Done. Pulled in your change where it is all inside a jsonloader func.